src/Microsoft.AspNetCore.Server.IISIntegration/IISMiddleware.cs

```diff
@@ -66,6 +74,17 @@ public IISMiddleware(RequestDelegate next, ILoggerFactory loggerFactory, IOption
                 return;
             }
 
+            // Handle shutdown from ANCM
+            if (HttpMethods.IsPost(httpContext.Request.Method) &&
+                string.Equals(ANCMShutdownRequestPath, httpContext.Request.Path.Value, StringComparison.OrdinalIgnoreCase) &&
+                string.Equals(ANCMShutdownEventHeaderValue, httpContext.Request.Headers[MSAspNetCoreEvent], StringComparison.Ordinal))
```

Comment:
ANCM is sending this so I think it's safe to use case-sensitive comparison. As a reference the token also uses case-sensitive compare.